### PR TITLE
warden support special kubecube scheme

### PR DESCRIPTION
**Pull request by tiancandevloper**

### CODE REVIEW

File: pkg/warden/reporter/report.go [hunk 1]
```diff
@@ -112,9 +112,19 @@ func (r *Reporter) do(info scout.WardenInfo) (*http.Response, error) {
...
-	url := fmt.Sprintf("https://%s%s", r.PivotCubeHost, "/api/v1/cube/scout/heartbeat")
+	// default, use https as scheme
+	uri, err := url.Parse(r.PivotCubeHost)
+	if err != nil {
+		return nil, err
+	}
+
+	uri.Path = "api/v1/cube/scout/heartbeat"
+	if len(uri.Host) == 0 {
+		uri.Scheme = "https"
```
Comment (weilaaa) [inline pkg/warden/reporter/report.go:123]: nit: change and move comment `use https scheme by default` here.

File: pkg/warden/reporter/report.go [hunk 2]
```diff
@@ -112,9 +112,19 @@ func (r *Reporter) do(info scout.WardenInfo) (*http.Response, error) {
...
+		uri.Host = r.PivotCubeHost
+	}
 
-	resp, err := r.Client.Post(url, "application/json", reader)
+	resp, err := r.Client.Post(uri.String(), "application/json", reader)
```
Comment (weilaaa) [inline pkg/warden/reporter/report.go:127]: Not sure if we should delete `&tls.Config{InsecureSkipVerify: true}` when it use http scheme, It doesn't matter?

Comment (tiancandevloper) [reply pkg/warden/reporter/report.go:127]: no problem

**Review by weilaaa — APPROVED** [review]

**Comment by weilaaa** [comment]
LGTM

**Pull request merged**